src/Client.php

```diff
@@ -116,6 +116,51 @@ public function getAllProducts(): array
         return $return;
     }
 
+	/**
+	 * Retrieves array of products.
+	 *
+	 * @param $chunkSize
+	 * @return Product[]
+	 */
+	public function getProductsByChunk( $chunkSize ): array
```

Comment:
please type hint the input argument to `int` and give it a default value of i guess 50. the return value should be `Generator`